File: AppTest/core/src/main/java/com/altamirano/fabricio/swipe/SwipeList.java
package com.altamirano.fabricio.swipe;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.BaseAdapter;
import android.widget.ListView;

public class SwipeList extends ListView {

    private SwipeActionAdapter mAdapter;

    public SwipeList(Context context) {
        super(context);
    }

    public SwipeList(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public SwipeList(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public void setSwipeActionListener(SwipeActionAdapter.SwipeActionListener listener) {
        this.mAdapter.setSwipeActionListener(listener);
    }

    public SwipeActionAdapter setAdapter(BaseAdapter adaper) {
        this.mAdapter = new SwipeActionAdapter(adaper);
        this.mAdapter.setListView(this);
        super.setAdapter(this.mAdapter);
        return this.mAdapter;
    }

    public SwipeActionAdapter addBackground(SwipeDirection key, int resId){
        this.mAdapter.addBackground(key, resId);
        return this.mAdapter;
    }

}

File: TestCore/core/src/main/java/com/altamirano/fabricio/core/swipe/SwipeList.kt
package com.altamirano.fabricio.core.swipe

import android.content.Context
import android.util.AttributeSet
import android.widget.BaseAdapter
import android.widget.ListView
import com.altamirano.fabricio.core.swipe.SwipeActionAdapter.SwipeActionListener

class SwipeList : ListView {
    private var mAdapter: SwipeActionAdapter? = null

    constructor(context: Context) : super(context) {}
    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {}
    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    ) {
    }

    fun setSwipeActionListener(listener: SwipeActionListener?) {
        mAdapter!!.setSwipeActionListener(listener)
    }

    fun setAdapter(adaper: BaseAdapter): SwipeActionAdapter {
        mAdapter = SwipeActionAdapter(adaper)
        mAdapter?.let {
            it.setListView(this)
            super.setAdapter(it)
        }
        return mAdapter!!
    }

    fun addBackground(key: SwipeDirection, resId: Int): SwipeActionAdapter? {
        mAdapter!!.addBackground(key, resId)
        return mAdapter
    }
}